sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -301,12 +318,235 @@ export class IER12ProcessingService {
         totalAssessedCost: assessmentData.totalAssessedCost,
         totalAssessmentNeed: assessmentData.totalAssessmentNeed,
         disbursementSentDate: disbursement.dateSent,
+        applicationEventCode: await this.getApplicationEventCode(
+          pendingAssessment,
+          disbursement,
+        ),
       };
       ier12Records.push(ier12Record);
     }
     return ier12Records;
   }
 
+  /**
+   * Get application event code (i.e current state of an application)
+   * @param assessment assessment of institutions.
+   * @param currentDisbursementSchedule current disbursement schedule.
+   * @returns application event code.
+   */
+  private async getApplicationEventCode(
+    assessment: StudentAssessment,
+    currentDisbursementSchedule: DisbursementSchedule,
+  ): Promise<ApplicationEventCode> {
+    const application = assessment.application;
+    switch (application.applicationStatus) {
+      case ApplicationStatus.Assessment:
+        return this.applicationEventCodeDuringAssessment(
+          application.applicationNumber,
+        );
+
+      case ApplicationStatus.Enrolment:
```

Comment:
The case condition and the method name doesn't match. `applicationEventCodeDuringEnrolmentAndCompleted`

Reply:
As mentioned in one of the previous comment, this method is used both during enrollment and completed status, https://github.com/bcgov/SIMS/pull/2374#discussion_r1353266461